e2e/app/modal/share-modal.ts

```diff
@@ -21,31 +21,57 @@ export default class ShareModal extends Modal {
   }
 
   async shareWithUser(userName: string, level: WorkspaceAccessLevel): Promise<void> {
-    const dropDownXpath = `${this.getXpath()}//*[@data-test-id="drop-down"]`;
+    const dropDownXpath = this.getXpath() + '//*[@data-test-id="drop-down"][.//clr-icon[@shape="plus-circle"]]';
     const waitForDropDown = async () => {
       await waitWhileLoading(this.page);
       await this.page.waitForXPath(dropDownXpath, { visible: true });
     };
+
     const waitForClose = async () => {
       await waitWhileLoading(this.page);
       await this.page.waitForXPath(dropDownXpath, { hidden: true });
     };
 
-    const searchBox = this.waitForSearchBox();
-    // Type in user email and append it with two whitespaces. Otherwise the select dropdown is not showing.
-    await searchBox.type(`${userName}  `, { delay: 20 });
+    const findCollaboratorAddIcon = () => {
+      return ClrIconLink.findByName(
+        this.page,
+        { type: ElementType.Icon, iconShape: 'plus-circle', containsText: userName, ancestorLevel: 2 },
+        this
+      );
+    };
 
-    await waitForDropDown(); // Needed for the dropdown
-    const addCollab = this.waitForAddCollaboratorIcon(userName);
-    await addCollab.click();
-    await waitForClose();
+    const pickUserRole = async () => {
+      const roleInput = await this.waitForRoleSelectorForUser(userName);
+      await roleInput.click();
 
-    const roleInput = await this.waitForRoleSelectorForUser(userName);
-    await roleInput.click();
+      const ownerOpt = await this.waitForRoleOption(level);
+      await ownerOpt.click();
+    };
 
-    const ownerOpt = await this.waitForRoleOption(level);
-    await ownerOpt.click();
+    let maxAttempts = 2;
+    // Try 2 times: enter email and find add icon.
+    const typeAndAddUser = async () => {
+      await this.waitForSearchBox().type(userName, { delay: 50 });
+      await waitForDropDown();
+      try {
+        const addIcon = findCollaboratorAddIcon();
+        await addIcon.click();
+        await waitForClose();
+        return;
+      } catch (err) {
+        // Click failed.
```

Comment:
I think it would be helpful to log exception here.

Reply:
I reworked code to search for add-icon as puppeteer type char by char.